client/src/pages/users/[company_id]/list/index.tsx

```diff
@@ -0,0 +1,170 @@
+import Button from '@/src/shared/components/Button';
+import InputField from '@/src/shared/components/InputField';
+import Modal from '@/src/shared/components/Modal/Modal';
+import EditIcon from '@/src/shared/icons/EditIcon';
+import XmarkIcon from '@/src/shared/icons/XmarkIcon';
+import useEditAddUser from '@/src/shared/hooks/useEditAddUser';
+import Select from '@/src/shared/components/Select';
+
+const UsersList: React.FC = () => {
+  const {
+    isAddModalOpen,
+    isEditModalOpen,
+    isConfirmationModalOpen,
+    confirmationMessage,
+    handleOpenAddModal,
+    handleOpenEditModal,
+    options,
+    role,
+    handleValueChange,
+    handleChange,
+    handleSubmitUser,
+    handleSubmitEdit,
+    handleConfirm,
+    handleDeleteUser,
+    formData,
+    setIsAddModalOpen,
+    setIsEditModalOpen,
+    setIsConfirmationModalOpen,
+    errorFistName,
+    errorLastName,
+    errorEmail,
+    errorPassword,
+    confirmPassword
+  } = useEditAddUser();
+
+  return (
+    <div>
+      <div>
+        <div className='ml-12 mb-12'>
+          <Button text='Add User' hover='hover:bg-blue-700' width='20' onClick={handleOpenAddModal} />
+          <div onClick={handleOpenEditModal} className='cursor-pointer'>
+            <EditIcon classname='text-blue-700 w-7 h-7 font-extrabold text-xl' />
+          </div>
+        </div>
+        <div>
+        </div>
+        <Modal isOpen={isAddModalOpen}>
+          <div className='flex justify-between relative mx-6' >
+            <div>
+              <h1 className='text-3xl mt-6 mb-9'>Add a new User</h1>
+            </div>
+            <div className='mt-8 cursor-pointer' onClick={() => { setIsAddModalOpen(false); }}>
+              <XmarkIcon />
+            </div>
+          </div>
+          <form onSubmit={handleSubmitUser}>
+            <div className='space-y-6 pb-5'>
+              <div>
+                <label className='ml-6 text-sm font-bold'>Role</label>
+                <div className='mx-6'>
+                  <Select options={options} value={role} eventHandler={handleValueChange} />
+                </div>
+              </div>
+              <div>
+                <label className='ml-6 text-sm font-bold'>First Name</label>
+                <div className='mx-6'>
+                  <InputField value={formData.firstName} eventHandler={handleChange} name="firstName" />

```

Comment:
Apply to all fields if necessary 

```suggestion
                  <InputField
                    value={formData.firstName}
                    eventHandler={handleChange}
                    name="firstName"
                    className={errorFistName.length > 0 ? 'border-red-500' : ''}
                  />

```